client/src/components/Header.tsx

```diff
@@ -30,16 +30,17 @@ const Header = () => {
   };
 
   return (
-    <Tabs>
+    <Tabs index={game.level}>
       <Flex bg="gray.200" w="100%" position="sticky" zIndex="1" p={4}>
         <TabList>
           <Tab onClick={() => game.jumpToLevel(0)}>Level 1</Tab>
-          <Tab onClick={() => game.jumpToLevel(1)}>Level 2</Tab>
-          <Tab onClick={() => game.jumpToLevel(2)}>Level 3</Tab>
-          <Tab onClick={() => game.jumpToLevel(3)}>Level 4</Tab>
-          <Tab onClick={() => game.jumpToLevel(4)}>Level 5</Tab>
+          <Tab isDisabled={!game.hasSeenLevel(0)} onClick={() => game.jumpToLevel(1)}>Level 2</Tab>
+          <Tab isDisabled={!game.hasSeenLevel(1)} onClick={() => game.jumpToLevel(2)}>Level 3</Tab>
+          <Tab isDisabled={!game.hasSeenLevel(2)} onClick={() => game.jumpToLevel(3)}>Level 4</Tab>
+          <Tab isDisabled={!game.hasSeenLevel(3)} onClick={() => game.jumpToLevel(4)}>Level 5</Tab>
```

Comment:
Be careful about this. `game.hasSeenLevel()` will only be called when the component mounts and may not reflect the current state of the variables. If this is producing the intended behaviour then go ahead.